src/Pi.h

```diff
@@ -92,6 +92,9 @@ class Pi {
 	static void SetMouseGrab(bool on);
 	static bool DoingMouseGrab() { return doingMouseGrab; }
 
+	static void EnableViewTransformationMode() { doingViewTransformation = true; }
+	static void DisableViewTransformationMode() { doingViewTransformation = false; }
+
```

Comment:
Please don't do this. We don't want to introduce any more functionality into `Pi.cpp` that isn't absolutely required, as it's already extremely painful to maintain this giant lump of singleton code.

In this case, please prefer using `ImGui::IsMouseDragging` and `ImGui::GetMouseDragDelta` (which aren't currently exposed to pigui) and pass the input into the systemview yourself. Among other reasons, this will reduce code duplication once mouse events become a first-class part of the input system, and it also doesn't add yet another global function to `Pi.cpp` that's only used for *one thing*.

Alternatively, you can expose `ImGui::CaptureMouseFromApp` (possibly with a different name and/or binding point), and use that functionality to keep from consuming mouse events while the user is rotating, though *stopping* rotating might be difficult.

Reply:
Done in d9ac37975b2f4161bc1350600ca22d3fe5cedf98!